### PR TITLE
Always consider fragments that specify a platform filter

### PR DESCRIPTION
**Pull request by laeubi**
A fragment can declare a Eclipse-PlatformFilter to be only installed on a given platform. If such fragment is currently wired to a host bundle, it is quite likely that fragment is needed but the host usually do not declare an Eclipse-ExtensibleAPI so we currently miss to add such fragment making the native code loading fails.

This now adds some code to store the platform filter and if one is given the fragment is always included.

Fix https://github.com/eclipse-pde/eclipse.pde/issues/1929

### CODE REVIEW

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/PDEAuxiliaryState.java [hunk 1]
```diff
@@ -97,6 +97,7 @@ static class PluginInfo {
 		String localization;
 		String bundleSourceEntry;
 		boolean exportsExternalAnnotations;
+		public String platformFilter;
```
Comment (merks) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/PDEAuxiliaryState.java:100]: I wonder why this is public but the others aren't?

Comment (laeubi) [reply ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/PDEAuxiliaryState.java]: That's what happening when using quick-fix to create the field and devloper not checking the results :-)

I just changed this, this might even can become a record these days, but don't want to change to much at once to keep the change count low.

**Comment by HannesWell** [comment]
> For me this is not really any heuristic, it is actually the reverse of Extensible-Api header, I never seen a case where one defines a Eclipse-PlatformFilter but don't want the fragment included, it is more the other way round that we take many efforts (e.g. additional `p2.inf`) to make sure the fragment is not badly missed.

All of this is in my opinion just a workaround because it's currently not possible to declare a requirement from a host to one of it's fragments and building the same in one reactor with Maven/Tycho.

Please see my answer in https://github.com/eclipse-pde/eclipse.pde/issues/1929#issuecomment-3212484648 

> > Furthermore this now affects all calls to the computation method that don't specify any options.
> 
> I now introduced a new option that is only activated in the `BundleLauncherHelper` so other callers should not be affected (even though I could not find it would harm anything on their call sites).

As written in the issue I'll probably reluctantly approve this, if this is only enabled for shrieked `JUnit Plugin Test` runtimes. But for general product's that's a no-go for me, as explicit means to include required fragments already exists e.g. for Products.

**Comment by laeubi** [comment]
> All of this is in my opinion just a workaround because it's currently not possible to declare a requirement from a host to one of it's fragments and building the same in one reactor with Maven/Tycho.

Actually native fragments (and Eclipse-PlatformFilter) are the "workaround", OSGi has standard means to declare and select appropriate native code by platform, anyways its an Eclipse Extension and PDE can/should support it.

"required fragments" is really something completely alien from OSGi point-of-view ... if it is required then it should be part of the bundle, if it is truly optional then it can be a fragment, but both does not really make sense.

> As written in the issue I'll probably reluctantly approve this, if this is only enabled for shrieked JUnit Plugin Test runtimes. But for general app/product's that's a no-go for me, for the mentioned reasons and as explicit means to include required fragments already exists e.g. for Products.

The whole purpose (and power) of "include required automatically" is that it works without specify it explicitly. SO of course one wants this for products and plain OSGi launches as well!

I really don't understand the concerns mentioned in the ticket they are talking about fragments that pull in more dependencies, about test fragments and so one none of this applies to the proposed solution here:

1. Native fragments only carry native code for the whole purpose of not shipping a full set of natives to get smaller bundles and have well defined meta-data carried with
2. They never have any additional dependencies
3. We perform additional work to make sure they are always installed
4. I have never seen a native test-fragment
